tests/object/test_object.py

```diff
@@ -64,6 +66,90 @@ def test_show(
 ]
 
 
+@mock.patch("snowflake.connector.connect")
+@pytest.mark.parametrize(
+    "object_type, input_scope, input_name",
+    [
+        ("schema", "database", "test_db"),
+        ("table", "schema", "test_schema"),
+        ("service", "compute-pool", "test_pool"),
+    ],
+)
+def test_show_with_scope(
+    mock_connector, object_type, input_scope, input_name, runner, mock_ctx
+):
+    ctx = mock_ctx()
+    mock_connector.return_value = ctx
+    result = runner.invoke(
+        ["object", "list", object_type, "--in", input_scope, input_name]
+    )
+    obj = OBJECT_TO_NAMES[object_type]
+    scope_obj = OBJECT_TO_NAMES[input_scope]
+    assert result.exit_code == 0, result.output
+    assert ctx.get_queries() == [
+        f"show {obj.sf_plural_name} like '%%' in {scope_obj.sf_name} {input_name}"
+    ]
+
+
+@mock.patch("snowflake.connector.connect")
+@pytest.mark.parametrize(
+    "object_type, input_scope, input_name, expected",
+    [
+        (
+            "table",
+            "invalid_scope",
+            "name",
+            "scope must be one of the following",
+        ),  # invalid scope label
+        (
+            "table",
+            "database",
+            "invalid name",
+            "scope name must be a valid identifier",
+        ),  # invalid scope identifier
+    ],
```

Comment:
nit: could consider moving the checking for correct error message to the unit test for _scope_validate and just ensure that if _scope_validate throws an error, result has an exit_code of 1